src/datasources/asset/components/AssetQueryEditor.tsx

```diff
@@ -13,73 +13,83 @@ import { defaultAssetSummaryQuery, defaultCalibrationForecastQuery, defaultListA
 import { ListAssetsQuery } from '../types/ListAssets.types';
 import { AssetSummaryQuery } from '../types/AssetSummaryQuery.types';
 
-type Props = QueryEditorProps<AssetDataSource, AssetQuery>;
+type Props = QueryEditorProps<AssetDataSource, AssetQuery, AssetDataSourceOptions>;
 
 export function AssetQueryEditor({ query, onChange, onRunQuery, datasource }: Props) {
-  const queryRef = useRef(query);
-  const onChangeRef = useRef(onChange);
-  const onRunQueryRef = useRef(onRunQuery);
-  const [queryType, setQueryType] = useState(AssetQueryType.ListAssets);
-
-  useEffect(() => {
-    queryRef.current = query;
-  }, [query]);
-
-  useEffect(() => {
-    onChangeRef.current = onChange;
-  }, [onChange]);
-
-  useEffect(() => {
-    onRunQueryRef.current = onRunQuery;
-  }, [onRunQuery]);
+  const [queryType, setQueryType] = useState(query.queryType as AssetQueryType);
+  const [assetFeatures] = useState<AssetFeatureToggles>({
+    assetList: datasource.instanceSettings.jsonData?.featureToggles?.assetList ?? AssetFeatureTogglesDefaults.assetList,
+    calibrationForecast: datasource.instanceSettings.jsonData?.featureToggles?.calibrationForecast ?? AssetFeatureTogglesDefaults.calibrationForecast,
+    assetSummary: datasource.instanceSettings.jsonData?.featureToggles?.assetSummary ?? AssetFeatureTogglesDefaults.assetSummary,
+  });
 
   const handleQueryChange = useCallback((value: AssetQuery, runQuery = false): void => {
```

Comment:
@Catalin-web I think it should be recomputed every time the query, onChange or onRunQuery variables changes. 

As per the official docs : `useCallback is a React Hook that lets you cache a function definition between re-renders.`

It's only the function definition that gets "recomputed" and not the execution. Since the `query` can change it is necessary to recompute the definition.

The `onChange, onRunQuery` are likely to never change but it's still a good practice to include dependencies. The linter also recommends this by default